### PR TITLE
Feature/cclow search filter

### PR DESCRIPTION
**Pull request by batraisk**

### CODE REVIEW

**Comment by tsubik** [comment]
:eyes:

File: app/controllers/concerns/cclow/filter_controller.rb [hunk 1]
```diff
@@ -14,7 +14,7 @@ def tags_options(taggable_type)
     end
 
     def filter_params
-      params.permit(:q, :fromDate, :recent, :ids, region: [], geography: [], tags: [])
+      params.permit(:q, :from_date, :to_date, :recent, :ids, region: [], geography: [], tags: [])
```
Comment (tsubik) [inline app/controllers/concerns/cclow/filter_controller.rb]: :+1:

File: app/javascript/components/pages/cclow/LitigationCases.js [hunk 2]
```diff
@@ -46,17 +49,45 @@ class LitigationCases extends Component {
...
   };
 
+  renderTimeRangeTags = (value) => (
+    <Fragment>
+      {value.from_date && (
+        <span key="tag-time-range-from" className="tag">
+          From {value.from_date}
+          <button
+            type="button"
+            onClick={() => this.timeRangeFilter.current.handelChange({from_date: null})}
```
Comment (tsubik) [inline app/javascript/components/pages/cclow/LitigationCases.js]: typo `handel` -> `handle` . Handel to na bazarze ;)

File: app/services/queries/cclow/litigation_query.rb [hunk 3]
```diff
@@ -48,7 +49,13 @@ def filter_by_tags
       def filter_by_from_date
         return scope unless params[:from_date].present?
 
-        scope.where('updated_at >= ?', params[:from_date])
+        scope.where('updated_at >= ?', Date.new(params[:from_date].to_i, 1, 1))
+      end
+
+      def filter_by_to_date
+        return scope unless params[:from_date].present?
```
Comment (tsubik) [inline app/services/queries/cclow/litigation_query.rb]: `to_date`

File: app/services/queries/cclow/target_query.rb [hunk 4]
```diff
@@ -38,6 +40,18 @@ def filter_by_geography
...
       end
 
+      def filter_by_from_date
+        return scope unless params[:from_date].present?
+
+        scope.where('updated_at >= ?', Date.new(params[:from_date].to_i, 1, 1))
+      end
+
+      def filter_by_to_date
+        return scope unless params[:from_date].present?
```
Comment (tsubik) [inline app/services/queries/cclow/target_query.rb]: `to_date`

File: app/assets/stylesheets/cclow/_filters.scss [hunk 5]
```diff
@@ -109,4 +113,50 @@ $shadow-cl: rgba(46,49,82,0.05);
...
+    .time-range-handle {
+      color: $red;
+      font-size: $size-7;
+      font-family: $family-sans-serif-bold;
+      position: absolute;
+      left: -50%;
+      text-align: center;
+      width: $time-slider-caption-width;
+
+      &.handel-0 {
```
Comment (tsubik) [inline app/assets/stylesheets/cclow/_filters.scss]: handle

File: app/assets/stylesheets/cclow/_filters.scss [hunk 6]
```diff
@@ -109,4 +113,50 @@ $shadow-cl: rgba(46,49,82,0.05);
...
+        top: -25px;
+        left: -($time-slider-caption-width / 2 - $time-slider-left-handel-width / 2);
+      }
+
+      &.handel-1 {
```
Comment (tsubik) [inline app/assets/stylesheets/cclow/_filters.scss]: handle

File: app/services/queries/cclow/legislation_query.rb [hunk 7]
```diff
@@ -48,7 +49,13 @@ def filter_by_tags
       def filter_by_from_date
         return scope unless params[:from_date].present?
 
-        scope.where('updated_at >= ?', params[:from_date])
+        scope.where('updated_at >= ?', Date.new(params[:from_date].to_i, 1, 1))
```
Comment (tsubik) [inline app/services/queries/cclow/legislation_query.rb]: I think the logic here will be more complicated unfortunately, not updated_at date

Comment (tsubik) [reply app/services/queries/cclow/legislation_query.rb]: I'm not sure what's the logic behind time filter, will try to find out.

**Comment by tsubik** [comment]
@batraisk about **from** and **to** date filter. Let's start with filtering by **last event** date instead of **updated_at** for **all** entities.

**Comment by tsubik** [comment]
Going to check this again

**Comment by tsubik** [comment]
Wasn't load more implemented? 
image

**Comment by batraisk** [comment]
I did not do this in this task @tsubik

**Comment by tsubik** [comment]
Hmm, I will try to add this quickly, as only 10 results are showing up on production

**Comment by tsubik** [comment]
Filters looks good, need more time to test them

**Pull request merged**